numpy/core/src/multiarray/mapping.c

```diff
@@ -727,8 +727,14 @@ prepare_index(PyArrayObject *self, PyObject *index,
         }
     }
     else if (used_ndim > PyArray_NDIM(self)) {
+	char error_message[50];
+        PyOS_snprintf(error_message, sizeof(error_message),
```

Comment:
I think it is better to use `PyErr_Format`:

1. You don't have to create a buffer `error_message` that may not be big enough and causes a failure in `PyOS_snprintf`
2. The code is actually shorter.

Most of the code uses the `PyErr_Format` idiom as well.